driver-sync/src/test/functional/com/mongodb/client/Fixture.java

```diff
@@ -110,11 +113,39 @@ public static MongoClientSettings.Builder getMongoClientSettings(final Connectio
     }
 
     public static ServerAddress getPrimary() throws InterruptedException {
-        getMongoClient();
-        List<ServerDescription> serverDescriptions = getPrimaries(((MongoClientImpl) mongoClient).getCluster().getDescription());
+        return getPrimary(getMongoClient());
+    }
+
+    /**
+     * Beware of a potential race condition hiding here: the primary you discover may differ from the one used by the {@code client}
+     * when performing some operations, as the primary may change.
+     */
+    public static ServerAddress getPrimary(final MongoClient client)
+            throws InterruptedException {
+        final Supplier<ClusterDescription> clusterDescriptionSupplier;
+        if (client instanceof MongoClientImpl) {
+            clusterDescriptionSupplier = () -> ((MongoClientImpl) client).getClusterDescription();
+        } else {
+            // com.mongodb.reactivestreams.client.syncadapter.SyncMongoClient
```

Comment:
I'm unclear why this is necessary, as `getClusterDescription` is defined on the `com.mongodb.client.MongoClient` interface not `MongoClientImpl` and is implement in `SyncMongoClient`.

Also see my comment on the new prose test, as it's also unclear to me why this method needs to be called again with its own MongoClient instance

Reply:
> getClusterDescription is defined on the com.mongodb.client.MongoClient interface not MongoClientImpl and is implement in SyncMongoClient

Thank you! I was confused by the original code of the method casting to `MongoClientImpl`. The implementation is now much simpler.

Reply:
Done.